src/components/hint/index.js

```diff
@@ -132,9 +133,19 @@ class Hint extends BaseComponent {
     }
   };
 
+  get targetLayout() {
```

Comment:
This method is called 54 times but the targetLayout only updated twice...

Reply:
it's just a getter to wrap the logic that returns the relevant `targetLayout`
cause each scenario uses a different value